resolwe/flow/tests/test_executors.py

```diff
@@ -344,31 +347,34 @@ def test_slug_colision(self):
         This checks for a bug in permission architecture that caused exception
         due to missing permission_group when slug colision occured.
         """
-        process = Process.objects.filter(slug="test-min").latest()
+        def pre_save(instance, add):
+            current_slug = slugs.pop(0)
+            if current_slug == IntegrityError:
+                raise IntegrityError("flow_data_slug")
+            instance.slug = current_slug
+            return current_slug
+
+        slugs = ["process_slug", "first_data_slug", IntegrityError, "second_data_slug"]        
+        slug_field_mock.side_effect = pre_save
+
+        expected_slug = slugs[0]
+        process = Process.objects.create(
+            name="Dummy process", contributor=self.contributor
+        )
+        self.assertEqual(process.slug, expected_slug)
+
+        expected_slug = slugs[0]
+        data1 = Data.objects.create(
+            name="data", process=process, contributor=self.contributor
+        )
+        self.assertEqual(data1.slug, expected_slug)
+
+        expected_slug = slugs[1]  # Skip IntegrityError
```

Comment:
And this `[3]`?

Reply:
Same as above. This one is 1 since there are two elements remaining: IntegrityError and the last slug.